pkg/destroy/powervs/cloudobjectstorage.go

```diff
@@ -0,0 +1,226 @@
+package powervs
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/pkg/errors"
+)
+
+const cosTypeName = "cos instance"
+const cosResourceID = "dff97f5c-bc5e-4455-b470-411c3edbe49c"
+
+// listCOSInstances lists COS service instances.
+func (o *ClusterUninstaller) listCOSInstances() (cloudResources, error) {
+	o.Logger.Debugf("Listing COS instances")
+
+	ctx, _ := o.contextWithTimeout()
+
+	options := o.controllerSvc.NewListResourceInstancesOptions()
+	options.SetResourceID(cosResourceID)
+	options.SetType("service_instance")
+
+	resources, _, err := o.controllerSvc.ListResourceInstancesWithContext(ctx, options)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list COS instances")
+	}
+
+	var foundOne = false
+
+	result := []cloudResource{}
+	for _, instance := range resources.Resources {
+		// Match the COS instances created by both the installer and the
+		// cluster-image-registry-operator.
+		if fmt.Sprintf("%s-cos", o.InfraID) == *instance.Name ||
+			fmt.Sprintf("%s-image-registry", o.InfraID) == *instance.Name {
+			foundOne = true
+			o.Logger.Debugf("listCOSInstances: FOUND %s %s", *instance.Name, *instance.GUID)
+			result = append(result, cloudResource{
+				key:      *instance.ID,
+				name:     *instance.Name,
+				status:   *instance.State,
+				typeName: cosTypeName,
+				id:       *instance.GUID,
+			})
+		}
+	}
+	if !foundOne {
+		o.Logger.Debugf("listCOSInstances: NO matching COS instance against: %s", o.InfraID)
+		for _, instance := range resources.Resources {
+			o.Logger.Debugf("listCOSInstances: only found COS instance: %s", *instance.Name)
+		}
+	}
+
+	return cloudResources{}.insert(result...), nil
+}
+
+func (o *ClusterUninstaller) findReclaimedCOSInstance(item cloudResource) (*resourcecontrollerv2.ResourceInstance, *resourcecontrollerv2.Reclamation) {
+	var (
+		getReclamationOptions *resourcecontrollerv2.ListReclamationsOptions
+		reclamations          *resourcecontrollerv2.ReclamationsList
+		response              *core.DetailedResponse
+		err                   error
+		reclamation           resourcecontrollerv2.Reclamation
+		getInstanceOptions    *resourcecontrollerv2.GetResourceInstanceOptions
+		cosInstance           *resourcecontrollerv2.ResourceInstance
+	)
+
+	getReclamationOptions = o.controllerSvc.NewListReclamationsOptions()
+
+	ctx, _ := o.contextWithTimeout()
+
+	reclamations, response, err = o.controllerSvc.ListReclamationsWithContext(ctx, getReclamationOptions)
+	if err != nil {
+		o.Logger.Debugf("Error: ListReclamationsWithContext: %v, response is %v", err, response)
+		return nil, nil
+	}
+
+	// ibmcloud resource reclamations --output json
```

Comment:
I see this command also lists the type called cloudcerts as well, how different this command from regular resource listing command?